fall18/p2/README.md

```diff
@@ -1,3 +1,149 @@
+
+**getArea(stateName)** 
+
+getArea function takes the name of one state as an argument and returns the size of this state area.
+
+```
+>>> import project
+>>> project.getArea("Wisconsin")
+>>> 65496.38
+```
+**getPopulation(stateName, year)**
+
+getPopulation function takes the name of one state and a year as arguments and returns the population of this state in this year.
+
+```
+>>> import project
+>>> project.getPopulation("Wisconsin", 2000)
+>>> 5363675
+```
+
+## Program Requirements
+You will write **at least four(4)** functions with the following names and behaviors:
+
+**getMaximumLand(stateName1, stateName2, stateName3)**
+
+Find the max value for land area among the three given area values which will be passed as parameters. To do this we can use the inbuilt [max()](https://docs.python.org/2/library/functions.html#max) function.
+
+Example:
+
+```
+>>> getMaximumLand("Wisconsin", "Iowa", "Minnesota")
+>>> 86935.83
+```
+
+**getMinimumPopulationDensity(stateName1, stateName2, stateName3, year)**
+
+Population density is measured as population per unit area. 
+
+The formula to calculate population density is:
+
+**Population Density = Population / Land Area (mi²)**
+
+For this function, you need to retrieve population and area values for the given states and calculate the density. We then find the min value of population density among the three values passed as parameters. To do this, we can use the inbuilt [min()](https://docs.python.org/2/library/functions.html#min) function.
+
+Example:
+
+```
+>>> getMinimumPopulationDensity("Wisconsin", "Iowa", "Minnesota", 2000)
+>>> 52.002450206414075
+```
+
+**predictPopulation(stateName, growthRate, yearA, yearB)**
+
+We use the following formula for population growth prediction in this question: <img src="Population.png" alt="drawing" width="100"/>. Here,
+
+ 1. <img src="Pa.png" alt="drawing" width="15"/> is the population for the specific state in year *a*.
+ 2. <img src="Pb.png" alt="drawing" width="15"/> is the predicted population for the specific state in year *b*, *b* > *a*. 
+ 3. *r* is the growth rate.
+ 4. *t* is the number of years between year *a* and the year *b*.
+
+The function should take in the values for the state name, state growth rate and year *a* and year *b*. It should return the value of the predicted population in year *b*. To implement this function, You can use exp function in Python [math library](https://docs.python.org/3/library/math.html).
+
+Example:
+
+```
+>>> predictPopulation("Wisconsin", 0.5, 2000, 2010)
+>>> 796039951.1495125
+```
+
+**calcGrowthRate(stateName, yearA, yearB)**
+
+Use the same formula as above to calculate the growth rate. To do this, you can use the log fucntion present in Python [math library](https://docs.python.org/3/library/math.html). 
```

Comment:
Maybe we should give them a bit more of a hint here.  For example, remind them that log(e**(rt)) == rt, and tell them they need to rearrange the above formula to isolate r.  Sadly, I don't think we can assume the freshmen in our classes will all be leaving high school comfortable working with logs.

Reply:
Yes, I too think that we should have this additional information here. @shenghaozou / @shrunambiar can you please add this?